app/src/org/commcare/models/database/user/models/EntityStorageCache.java

```diff
@@ -115,7 +115,9 @@ public void invalidateCaches(Collection<Integer> recordIds) {
         List<Pair<String, String[]>> whereParamList = TableBuilder.sqlList(recordIds);
         int removed = 0;
         for (Pair<String, String[]> querySet : whereParamList) {
-            removed += db.delete(TABLE_NAME, COL_CACHE_NAME + " = '" + this.mCacheName + "' AND " + COL_ENTITY_KEY + " IN " + querySet.first, querySet.second);
+            removed += db.delete(TABLE_NAME, COL_APP_ID + " = '" + mAppId + "' AND " +
```

Comment:
Somewhat counterintuitively, I don't think this specific call actually should be app restricted.

If the case model changes, it should invalidate caches around that case model regardless of which app produced them.

Reply:
What about the respective singular method - `invalidateCache` . Is it fine to incorporate appId in that. I can see that whie doing bulk processing we use `invalidateCaches` instead of `invalidateCache` but couldn't really find any code which indicate that `invalidateCaches` gets called in case of case model changes. Also a bit unclear what case model refers to in this case specifically.

Reply:
I think in general any time we are invalidating a cache due to a change to a case record we should *not* restrict by appId. 

I was under the impression that this gets called after a case is changed, and that the recordId is the row Id of a case in the db, is that not correct?

Reply:
Hmm. I agree that we don't need to restrict these calls by appID as we are querying on case id which would be a sufficient condition to get to a specific row.  Though this gets called from `BulkElementParser::parse` and I am not sure whether it's specically related to case model change.